tests/unittest/test_requests.py

```diff
@@ -567,10 +567,9 @@ def test_stream_empty_body(server):
 def test_stream_incomplete_read(server):
     with requests.Session() as s:
         url = str(server.url.copy_with(path="/incomplete_read"))
-        with pytest.raises(requests.RequestsError) as e:
```

Comment:
The nested style is preferred to show the two different environment we are creating. It's not like we are opening two files here.